codice/ddf
# DDF-6360 Upgrade to CXF 3.4.0

#### What does this PR do?
Upgrades to CXF 3.4.0, and along with it upgrades a bunch of other dependencies, including:
* OpenSAML -> 3.4.5
* WSS4J -> 2.3.0
* Santuario (xmlsec) -> 2.2.0
* JAXB API -> Jakarta JAXB API
* Sun JAXB impl to Glassfish JAXB impl
* Javax Annotation API -> Jakarta Annotation API
* JavaBeans Activation Framework API -> Jakarta Activation API

#### Who is reviewing it? 

@bakejeyner @blen-desta @emmberk @garrettfreibott @stustison

#### Select relevant component teams: 
@codice/security 

#### How should this be tested?

The major issues I ran into when upgrading were very slow installation times and JAXB errors. With that in mind, verify that:. With that in mind, verify that:

* Installation is not abnormally slow
* Anything using JAXB still works (XML marshalling, unmarshalling)
  * Adding and querying sources: CSW & OpenSearch (loopback is fine) and WFS (I used http://geomap.reteunitaria.piemonte.it/ws/gsareprot/rp-01/areeprotwfs/wfs_gsareprot_1, which is WFS 1.1.0 and just tested that the source is shown as active)
  * XML transformer: ingest an XML document with geometry, then download it via the xml transformer


#### Any background context you want to provide?
This is part of an ongoing effort to reduce the CVE level in DDF

#### What are the relevant tickets?
Fixes: #6360

#### Screenshots


#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
